docs/architecture/subscriptions.md

```diff
@@ -0,0 +1,214 @@
+# Subscriptions
+
+A **subscription** function represents a dependency your app has on some external process. As with [effects](actions.md#effects), subscriptions deal with impure, asynchronous interactions with the outside world in a safe, pure, and immutable way. They are a streamlined way of responding to events happening outside our application such as time or location changes. They handle resource management for us that we would otherwise need to worry about like adding and removing event listeners, closing connections, etc.
```

Comment:
I agree that "paying attention to" is better than "dependency". I want to maintain the style of defining things on an individual basis, so I'm thinking of going with a variation of your suggestion:

"A **subscription** function represents an external process that your app needs to continually pay attention to."

How's that ?

Reply:
Maybe instead of defining it, just say what it's good for?

Here's some copy, if you can improve it, even better:

> Subscriptions allow you to listen to external events such as incoming WebSocket messages, clock tick events, mouse, keyboard, window events, geolocation changes, and more. Think of them as declarative events.

Reply:
I agree that highlighting specific use cases is a good thing.  I also think giving an official definition is important not only to set basic expectations but to prove that it **can** be defined.